qiskit_ibm_runtime/sampler.py

```diff
@@ -180,14 +181,25 @@ def run(
         Raises:
             ValueError: If the input values are invalid.
         """
-        if isinstance(circuits, Iterable) and not all(
-            isinstance(inst, QuantumCircuit) for inst in circuits
-        ):
-            raise ValueError(
-                "The circuits parameter has to be instances of QuantumCircuit."
-            )
+        if isinstance(circuits, QuantumCircuit):
+            circuits = [circuits]
+
+        for circ in circuits:
+            if not isinstance(circ, QuantumCircuit):
+                raise ValueError(
+                    "The circuits parameter has to be instances of QuantumCircuit."
+                )
```

Comment:
Minor comment, and I don't know what the standard in qiskit-ibm-provider is, but in terra we always assume the type hint is satisfied, so this check could be omitted since by the type hint only `QuantumCircuit`s are allowed 🙂

Reply:
Yes, but I just left the original code as is.
https://github.com/Qiskit/qiskit-ibm-runtime/blob/af308caeb7c261a1fb1a7ca7a45c49f55df02215/qiskit_ibm_runtime/sampler.py#L186-L188